common/lib/capa/capa/safe_exec/helpers.py

```diff
@@ -0,0 +1,77 @@
+"""
+Helper methods related to safe exec.
+"""
+
+import requests
+import json
+
+from django.conf import settings
+from edx_toggles.toggles import SettingDictToggle
+
+# .. toggle_name: ENABLE_CODEJAIL_REST_SERVICE
+# .. toggle_implementation: DjangoSetting
+# .. toggle_default: False
+# .. toggle_description: Set this to True if you want to run Codejail code using
+#   a separate VM or container and communicate with edx-platform using REST API.
+# .. toggle_use_cases: tutor
+# .. toggle_creation_date: 2021-08-19
+# .. toggle_target_removal_date: None
+# .. toggle_warnings: 
+# .. toggle_tickets:
+ENABLE_CODEJAIL_REST_SERVICE = SettingDictToggle(
+    "FEATURES", "ENABLE_CODEJAIL_REST_SERVICE", default=False, module_name=__name__
+)
+
+def is_codejail_rest_service_enabled():
+    return ENABLE_CODEJAIL_REST_SERVICE.is_enabled()
+
+def get_codejail_rest_service_endpoint():
+    return  "".join([
+        settings.CODE_JAIL_REST_SERVICE_HOST,
+        "/api/v0/code-exec"
+        ])
+
+def send_safe_exec_request(data, extra_files):
+    """
+    Sends a request to a codejail api service forwarding required code and files.
+    Arguments:
+        data: Dict containing code and othe parameters 
+            required for jailed code execution.
+        extra_files: python_lib.zip file containing extra files 
+            required by the codejail execution.
+    Returns:
+        Response received from codejsail api service
+    """
+    globals_dict = data["globals_dict"]
+
+    codejail_service_endpoint = get_codejail_rest_service_endpoint()
+    payload = json.dumps(data)
+
+    try:
+        response = requests.post(
+            codejail_service_endpoint,
+            files=extra_files,
+            data={'payload': payload},
+            timeout=(settings.CODE_JAIL_REST_SERVICE_CONNECT_TIMEOUT, settings.CODE_JAIL_REST_SERVICE_READ_TIMEOUT)
+        )
+
+        response_json = response.json()
+
+        emsg = response_json["emsg"]
+        exception =  None
+
+        if emsg:
+            exception_msg = ". ".join([
+                    emsg,
+                    "For more information check Codejail Service logs."
+                ])
+            
+            exception = SafeExecException(emsg)
+
+        globals_dict.update(response_json["globals_dict"])
```

Comment:
```suggestion
        globals_dict.update(response_json.get("globals_dict"))
```
Indeed, what happens if the response does not contain a "globals_dict" key? Maybe validation is required to make sure "globals_dict" is returned and it contains proper data